tidy3d/components/medium.py

```diff
@@ -1085,3 +1094,171 @@ def plot(self, freqs: float, ax: Ax = None) -> Ax:
+        "xx-component of the corresponding 3D medium.",
+        discriminator=TYPE_TAG_STR,
+    )
+
+    tt: IsotropicMediumType = pd.Field(
+        ...,
+        title="TT Component",
+        description="Medium describing the tt-component of the diagonal permittivity tensor. "
+        "If the 2D material is normal to the y-axis, for example, then this determines the "
+        "zz-component of the corresponding 3D medium.",
+        discriminator=TYPE_TAG_STR,
+    )
+
+    def volumetric_equivalent(
+        self, axis: int, neighbors: Tuple[MediumType3D, MediumType3D], dls: Tuple[float, float]
+    ) -> AnisotropicMedium:
+        """Produces a 3D volumetric equivalent medium. The new medium has thickness equal to
+        the average of the ``dls`` in the ``axis`` direction.
+        The ss and tt components of the 2D material are mapped in order onto the xx, yy, and
+        zz components of the 3D material, excluding the ``axis`` component. The conductivity
+        and residues (in the case of a dispersive 2D material) are rescaled by 1/dl.
+        The neighboring media ``neighbors`` enter in as a background for the resulting
+        volumetric equivalent.
+
+
+        Parameters
+        ----------
+        axis : int
+            Index (0, 1, or 2 for x, y, or z respectively) of the normal direction to the
+            2D material.
+        neighbors : Tuple[MediumType3D, MediumType3D]
+            The neighboring media on either side of the 2D material.
+        dls : Tuple[float, float]
+            Each dl represents twice the thickness of the desired volumetric model on the
+            respective side of the 2D material.
+
+        Returns
+        -------
+        AnisotropicMedium
+            The 3D material corresponding to this 2D material.
+        """
+
+        def get_component(med: MediumType3D, comp: int) -> IsotropicMediumType:
+            """Extract the `comp` component of `med`."""
+            if isinstance(med, AnisotropicMedium):
+                return getattr(med, ["xx", "yy", "zz"][comp])
+            return med
+
+        def weighted_avg(meds: List[IsotropicMediumType], weights: List[float]) -> PoleResidue:
+            """Average `meds` with weights `weights`."""
+            eps_inf = 1
+            poles = []
+            for (med, weight) in zip(meds, weights):
+                eps_inf += weight * (med.eps_model(inf) - 1)
+                if isinstance(med, DispersiveMedium):
+                    poles += [(a, weight * c) for (a, c) in med.pole_residue.poles]
+                elif med.conductivity != 0:
+                    poles.append((0, weight * med.conductivity / (2 * EPSILON_0)))
+            return PoleResidue(eps_inf=np.real(eps_inf), poles=poles)
+
+        def get_background(comp: int) -> PoleResidue:
+            """Get the background medium appropriate for the `comp` component."""
+            meds = [get_component(neighbors[i], comp) for i in range(2)]
```

Comment:
it's slightly better to pass things as keyword arguments if you have space.
```py
get_component(med=neighbors[i], comp=comp)
```
this is more robust in case there is ever a change in the argument order.
Can you also do this in other instances where this occurs in this method?